src/lib/stores/auth-store.ts

```diff
@@ -36,6 +40,13 @@ export const createAuthStore = (initState: AuthState = defaultInitState) => {
           logout: () => {
             set(defaultInitState);
           },
+          setIsCoach: (coachId) => {
+            var userId = get().userId;
+            set({ isCoach: !!(userId && coachId && userId == coachId) });
```

Comment:
I think it still makes sense to store `isCoach` as a property in this store, but perhaps it makes sense to move this logic into a function as part of the UserSession type, in src/types/user-session.ts? That seems to me to be grouping like things together where one would expect them. Let me know what you think.

Reply:
Can you elaborate on your thinking around this? In my mind the logic belongs here because the `isCoach` concept is a part of the `AuthState` interface because it's a derived virtual state that is dependent on the coaching relationship context as opposed to a more static property of the `UserSession` (Ex. `first_name`).

I want to make sure I understand where you're coming from though as this is still fairly new to me.

Reply:
Yeah I think it's fine for now, I was over-thinking this one.